src/components/dialogs/two-windings-transformer/phase-tap-changer-pane.js

```diff
@@ -0,0 +1,563 @@
+import { REGULATION_MODES } from '../../network/constants';
+import Papa from 'papaparse';
+import makeStyles from '@mui/styles/makeStyles';
+import { PHASE_TAP } from './two-windings-transformer-creation-dialog';
+import { CreateRuleDialog } from './create-rule-dialog';
+import { ImportRuleDialog } from './import-rule-dialog';
+
+const useStyles = makeStyles((theme) => ({
+    center: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    tableCell: {
+        fontSize: 'small',
+        cursor: 'initial',
+        padding: theme.spacing(1.25),
+        display: 'flex',
+        '&:before': {
+            content: '""',
+            position: 'absolute',
+            left: theme.spacing(0.5),
+            right: theme.spacing(0.5),
+            bottom: 0,
+            borderBottom: '1px solid ' + theme.palette.divider,
+        },
+    },
+}));
+
+const PhaseTapChangerPane = (props) => {
+    const {
+        formValues,
+        setFormValues,
+        phaseTapRows,
+        handlePhaseTapRows,
+        phaseTapChangerEnabled,
+        phaseTapChangerEnabledField,
+        regulationModeField,
+        regulationMode,
+        currentLimiterRegulatingValueField,
+        flowSetPointRegulatingValueField,
+        targetDeadbandField,
+        regulatingTerminalField,
+        lowTapPositionField,
+        lowTapPosition,
+        highTapPositionField,
+        highTapPosition,
+        tapPositionField,
+        regulatingField,
+    } = props;
+
+    const classes = useStyles();
+
+    const intl = useIntl();
+
+    const [lineEdit, setLineEdit] = useState(undefined);
+
+    const [openCreateRuleDialog, setOpenCreateRuleDialog] = useState(false);
+
+    const [openImportRuleDialog, setOpenImportRuleDialog] = useState(false);
+
+    const isLineOnEditMode = useCallback(
+        (rowData) => {
+            return lineEdit && rowData.rowIndex === lineEdit.id;
+        },
+        [lineEdit]
+    );
+
+    const generateNewTapData = (index) => {
+        return {
+            key: index,
+            tap: index,
+            resistance: 0,
+            reactance: 0,
+            conductance: 0,
+            susceptance: 0,
+            ratio: 1,
+            alpha: 0,
+            isEdited: false,
+        };
+    };
+
+    const generateTapRows = () => {
+        let tempRows = [];
+        const rowNumber =
+            highTapPosition - lowTapPosition > 0
+                ? highTapPosition - lowTapPosition + 1
+                : 0;
+
+        if (
+            phaseTapRows.length !== rowNumber &&
+            parseInt(lowTapPosition, 10) >= 0 &&
+            parseInt(highTapPosition, 10) >= 0 &&
+            !isNaN(parseInt(lowTapPosition, 10)) &&
+            !isNaN(parseInt(highTapPosition, 10))
+        ) {
+            for (let i = lowTapPosition; i <= highTapPosition; i++) {
+                tempRows.push(generateNewTapData(i));
+            }
+            let editedRows = Array.from(
```

Comment:
It is in case the user resizes the tap rows array to a size inferior than the index of a previously edited row. 
In this case we discard the edited row in hand